inference/core/workflows/core_steps/fusion/detections_stitch/v1.py

```diff
@@ -0,0 +1,252 @@
+        re_aligned_predictions = []
+        for detections in predictions:
+            detections_copy = deepcopy(detections)
+            resolution_wh = retrieve_crop_wh(detections=detections_copy)
+            offset = retrieve_crop_offset(detections=detections_copy)
+            detections_copy = manage_crops_metadata(
+                detections=detections_copy,
+                offset=offset,
+                parent_id=reference_image.parent_metadata.parent_id,
+            )
+            re_aligned_detections = move_detections(
+                detections=detections_copy,
+                offset=offset,
+                resolution_wh=resolution_wh,
+            )
+            re_aligned_predictions.append(re_aligned_detections)
+        overlap_filter = choose_overlap_filter_strategy(
+            overlap_filtering_strategy=overlap_filtering_strategy,
+        )
+        merged = sv.Detections.merge(detections_list=re_aligned_predictions)
+        if overlap_filter is OverlapFilter.NONE:
+            return {"predictions": merged}
+        if overlap_filter is OverlapFilter.NON_MAX_SUPPRESSION:
+            return {"predictions": merged.with_nms(threshold=iou_threshold)}
+        return {"predictions": merged.with_nmm(threshold=iou_threshold)}
+
+
+def retrieve_crop_wh(detections: sv.Detections) -> Optional[Tuple[int, int]]:
+    if len(detections) == 0:
+        return None
+    if PARENT_DIMENSIONS_KEY not in detections.data:
+        raise RuntimeError(
+            f"Dimensions for crops is expected to be saved in data key {PARENT_DIMENSIONS_KEY} "
+            f"of sv.Detections, but could not be found. Probably block producing sv.Detections "
+            f"lack this part of implementation or has a bug."
+        )
+    return (
+        detections.data[PARENT_DIMENSIONS_KEY][0][1].item(),
+        detections.data[PARENT_DIMENSIONS_KEY][0][0].item(),
+    )
+
+
+def retrieve_crop_offset(detections: sv.Detections) -> Optional[np.ndarray]:
+    if len(detections) == 0:
+        return None
+    if PARENT_COORDINATES_KEY not in detections.data:
+        raise RuntimeError(
+            f"Offset for crops is expected to be saved in data key {PARENT_COORDINATES_KEY} "
+            f"of sv.Detections, but could not be found. Probably block producing sv.Detections "
+            f"lack this part of implementation or has a bug."
+        )
+    return detections.data[PARENT_COORDINATES_KEY][0][:2].copy()
+
+
+def manage_crops_metadata(
+    detections: sv.Detections,
+    offset: Optional[np.ndarray],
+    parent_id: str,
+) -> sv.Detections:
+    if len(detections) == 0:
+        return detections
+    if offset is None:
+        raise ValueError(
+            "To process non-empty detections offset is needed, but not given"
+        )
+    if SCALING_RELATIVE_TO_PARENT_KEY in detections.data:
+        scale = detections[SCALING_RELATIVE_TO_PARENT_KEY][0]
+        if abs(scale - 1.0) > 1e-4:
+            raise ValueError(
```

Comment:
what s an example of this happening?

Reply:
if you bring operation of bboxes scaling in between model and stitch, which you should not be doing, but you can through UQL